java-checks/src/test/files/checks/ConstructorCallingOverridableCheck.java

```diff
@@ -138,3 +138,68 @@ void foo(Easy easy) {
     T qix();
   }
 }
+
+abstract class E {
+  private final MyInterface myInterface1;
+  private final MyInterface myInterface2;
+
+  public E() {
+    myInterface1 = new MyInterface() {
+      @Override
+      public String foo() throws Exception {
+        return doInBackground(); // Compliant
+      }
+    };
+    myInterface2 = () -> doInBackground(); // Compliant
+  }
+
+  protected abstract String doInBackground() throws Exception;
+
+  interface MyInterface {
+    String foo() throws Exception;
+  }
+}
+
+class F {
+  public F(){
+    yaml = memoize(() -> loadYamlConfig("_config.yml")); // Compliant
+
+    data = memoize(() -> getResourceList() // Compliant
+      .stream()
+      .filter(path -> path.startsWith("_data/"))
+      .collect(toMap(Site::nameWithoutExtension, this::readYaml))
+    );
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> loadYamlConfig(String configFile) {
+    Path configPath = Paths.get(configFile);
```

Comment:
You don't need to fill the body of that method, you just need it to use it as lambda.